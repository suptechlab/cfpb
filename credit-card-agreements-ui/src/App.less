@import (less) "~cf-core/src/cf-core.less";
@import (less) "~cf-icons/src/cf-icons.less";
@import (less) "~cf-buttons/src/cf-buttons.less";
@import (less) "~cf-forms/src/cf-forms.less";
@import (less) "~cf-grid/src/cf-grid.less";
@import (less) "~cf-layout/src/cf-layout.less";
@import (less) "~cf-typography/src/cf-typography.less";
@import (less) "~cf-pagination/src/cf-pagination.less";
@import (less) "~cf-expandables/src/cf-expandables.less";
@import (less) "~cf-tables/src/cf-tables.less";

@import (less) "base.less";

// App-wide definitions

.layout-row {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}

.layout-column {
    display: flex;
    flex-direction: column;
}

.flex-fixed {
    flex: none;
}

.flex-all {
    flex: 1 0 auto;
}

.lt-ie10 {
    .layout-row {
        display: table;
    }

    .layout-column {
        display: table;
    }

    .flex-fixed {
        float: left;
    }

    .flex-all {
        float: left;
    }
}

.body-copy {
    font-size: 16px;
}

.cols {
    .grid_nested-col-group();
}

.col:before {
    display: block;

    width: 100%;
    content: "";
}

.col-1 {
    .grid_column( 1 );
}

.col-2 {
    .grid_column( 2 );
}

.col-3 {
    .grid_column( 3 );
}

.col-4 {
    .grid_column( 4 );
}

.col-6 {
    .grid_column( 6 );
}

.col-8 {
    .grid_column( 8 );
}

.col-9 {
    .grid_column( 9 );
}

.col-10 {
    .grid_column( 10 );
}

.col-12 {
    .grid_column( 12 );
}

// Overrides

.content__1-3 {
    margin-bottom: @grid_gutter-width * 2;

    .content_wrapper {
        padding-left: 0;
        padding-right: 0;

        max-width: initial;
    }

    .content_sidebar {
        width: 420px;
        padding: 0;
        border: 0;
        margin: 0;
    }

    .content_main {
        width: ~"calc(100% - 420px)";
        padding: 0;
        border: 0;

        &:after {
            border-left: 0;
        }
    }
}

@media @tablet {
    .content__1-3 {
        .content_sidebar {
            width: 305px;
        }

        .content_main {
            width: ~"calc(100% - 305px)";
        }
    }
}

@media @phone {
    .content__1-3 {
        margin-bottom: 0;

        .content_sidebar {
            width: 0;
            display: none;
        }

        .content_main {
            width: 100%;
        }
    }
}

body {
    a, .a-btn__link {
        font-family: @webfont-medium;

        color: @link-text;

        &:hover {
            color: @link-text-hover;
        }
    }

    h4 {
        font-size: 18px;
    }

    #root {
      max-width: 1200px;
      margin: 0 auto;
    }

}

::-ms-clear {
    display: none;
}
