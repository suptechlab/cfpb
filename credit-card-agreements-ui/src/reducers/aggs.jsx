import { COMPLAINTS_RECEIVED } from '../constants'

/* eslint-disable camelcase */

export const defaultAggs = {
  company: [],
  company_public_response: [],
  company_response: [],
  consumer_consent_provided: [],
  consumer_disputed: [],
  issue: [],
  product: [],
  state: [],
  submitted_via: [],
  tag: [],
  timely: [],
  zip_code: []
}

/* eslint-enable camelcase */

export default ( state = defaultAggs, action ) => {
  switch ( action.type ) {
    case COMPLAINTS_RECEIVED: {
      const programs = action.data.data
      const result = { ...state }

      programs.forEach( program => {
        result[program.slug] = program
      } )

      return result
    }

    default:
      return state
  }
}
