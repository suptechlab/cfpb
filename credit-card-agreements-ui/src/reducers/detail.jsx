import {
  COMPLAINT_DETAIL_FAILED, COMPLAINT_DETAIL_RECEIVED
} from '../constants'

const defaultDetail = {
  data: {},
  error: ''
}

export default ( state = defaultDetail, action ) => {
  switch ( action.type ) {
    case COMPLAINT_DETAIL_RECEIVED:
      return {
        data: action.data,
        error: ''
      }

    case COMPLAINT_DETAIL_FAILED:
      return {
        data: {},
        error: action.error
      }

    default:
      return state
  }
}
