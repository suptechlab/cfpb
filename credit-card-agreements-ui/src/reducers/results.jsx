/* eslint-disable camelcase */

import {
  API_CALLED, COMPLAINTS_FAILED, COMPLAINTS_RECEIVED
} from '../constants'

const defaultResults = {
  activeCall: '',
  doc_count: 0,
  error: '',
  lastUpdated: null,
  lastIndexed: null,
  hasDataIssue: false,
  isDataStale: false,
  isNarrativeStale: false,
  isLoading: false,
  items: [],
  total: 0
}

export default ( state = defaultResults, action ) => {
  switch ( action.type ) {
    case API_CALLED:
      return {
        ...state,
        activeCall: action.url,
        isLoading: true
      }

    case COMPLAINTS_RECEIVED: {
      const items = action.data.data

      return {
        ...state,
        activeCall: '',
        doc_count: items.length,
        error: '',
        isLoading: false,
        items: items
      }
    }

    case COMPLAINTS_FAILED:
      return {
        ...defaultResults,
        error: action.error
      }

    default:
      return state
  }
}
