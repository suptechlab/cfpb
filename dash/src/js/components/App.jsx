// var Footer = require('./Footer.jsx');
var Header = require('./Header.jsx');
var React = require('react');
var Home = require('./Home.jsx');
var Crumbs = require('./Crumbs.jsx');
var _ = require('lodash');
var Router = require('react-router');
var teamStore = require('../stores/teamStore');
var teamDetailStore = require('../stores/teamDetailStore');
var userStore = require('../stores/userStore');
var loggedInUserStore = require('../stores/loggedInUserStore');

var App = React.createClass({
  stores: {
    teamStore: teamStore,
    users: userStore,
    loggedInUser: loggedInUserStore,
    teamDetails: teamDetailStore
  },
  getAppState: function() {
    var state = {};
    _.each(this.stores, function( store, storeName ) {
      state[storeName] = store.getState()
    });
    return state;
  },
  getInitialState: function() {
    return this.getAppState();
  },
  isReady: function() {
    return this.state.users.length &&
      this.state.loggedInUser.get('name') &&
      true;
  },
  componentDidMount: function() {
    var that = this;
    _.each(this.stores, function( store ) {
      store.onChange(that._onChange, that);
    })
  },
  componentWillUnmount: function() {
    var that = this;
    _.each(this.stores, function(store) {
      store.off(null, null, that)
    })
  },
  _onChange: function() {
    this.setState(this.getAppState())
  },
  render: function() {
    var Body = <div></div>;

    if (this.isReady()) {
      Body = (
        <div>
          <Router.RouteHandler {...this.state} />
        </div>
      );
    } else if (this.state.loggedInUser.isLoggedIn()) {
      Body = <div>Loading...</div>;
    }

    return (
      <div className="app">
        <Header loggedInUser={this.state.loggedInUser} />
        <main className="content" id="main" role="main">
          <div className="content_bar"></div>
          <div className="content_wrapper">
            <div className="content_main">
              <Crumbs />
              {Body}
            </div>
          </div>
        </main>
      </div>
    );
  }

});

module.exports = App;
