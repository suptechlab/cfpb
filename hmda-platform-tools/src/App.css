@font-face {
  font-family: SourceSansPro;
  src: url(./fonts/sourcesanspro-regular-webfont.woff);
}
@font-face {
  font-family: SourceSansProBold;
  src: url(./fonts/sourcesanspro-bold-webfont.woff);
}
@font-face {
  font-family: SourceSansProItalic;
  src: url(./fonts/sourcesanspro-italic-webfont.woff);
}

*, *::before, *::after {
  box-sizing: inherit;
}

html {
  box-sizing: border-box;
  font-family: SourceSansPro;
  font-size: 10px;
  min-height: 100%;
  position: relative;
}
body {
  margin: 0 0 350px;
  padding: 0;
  font-family: SourceSansPro;
  font-size: 1.7rem;
}
.skipnav {
  background: transparent;
  color: #212121;
  left: 0;
  padding: 1rem 1.5rem;
  position: absolute;
  top: -4.2rem;
  transition: all 0.2s ease-in-out;
  z-index: 100;
}

a {
  color: #0071bc;
  text-decoration: underline;
}
a:visited {
  color: #4c2c92;
}
a:hover,
a:active {
  color: #205493;
}

p, li {
  line-height: 1.5;
}

li {
  margin-bottom: 0.5em;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: SourceSansProBold;
  clear: both;
  margin-top: 1.5em;
  margin-bottom: 0.5em;
  line-height: 1.3;
}

ol, ul {
  padding-left: 1.94em;
}

h1+ul, h1+ol, h2+ul, h2+ol, h3+ul, h3+ol, h4+ul, h4+ol, h5+ul, h5+ol, h6+ul, h6+ol, p+ul, p+ol {
  margin-top: 0;
}

hr {
  border-bottom: none;
  border-top: 1px solid #aeb0b5;
}

.font-lead {
  font-size: 2rem;
  line-height: 1.7;
  margin-top: 0;
  max-width: 77rem;
  color: #5b616b;
}

.text-small {
  font-size: 1.3rem;
}

#main-content .header {
  border-bottom: none;
}

#main-content .grid {
  align-items: stretch;
  display: flex;
  flex-wrap: wrap;
  padding: 0;
}

#main-content .grid .item {
  flex: 1 1 450px;
  margin: 3rem 0;
  padding-right: 75px;
}

#main-content h1 {
  font-size: 4rem;
  margin-top: 1.5em;
}

.grid {
  padding-right: 1.5rem;
  padding-left: 1.5rem;
  margin: 0 auto;
  max-width: 1040px;
}

@media screen and (min-width: 600px) {
  .grid {
    padding-right: 3rem;
    padding-left: 3rem;
  }
}
