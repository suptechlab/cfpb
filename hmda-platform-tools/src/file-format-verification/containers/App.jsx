import React, { Component } from 'react'
import { connect } from 'react-redux'

export class AppContainer extends Component {
  render() {
    return <div className="AppContainer">{this.props.children}</div>
  }
}

export default connect()(AppContainer)
