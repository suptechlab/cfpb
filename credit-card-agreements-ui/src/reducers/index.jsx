import aggs from './aggs'
import { combineReducers } from 'redux'
import detail from './detail'
import modal from './modal'
import query from './query'
import results from './results'

export default combineReducers( {
  aggs,
  detail,
  modal,
  query,
  results
} )
