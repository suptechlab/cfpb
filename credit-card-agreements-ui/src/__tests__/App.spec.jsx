import { App, DetailComponents } from '../App';
import configureMockStore from 'redux-mock-store'
import { defaultQuery } from '../reducers/query'
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux'
import React from 'react';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk'

describe('initial state', () => {
  it('renders without crashing', () => {
    defaultQuery.searchText = 'foo';

    const target = renderer.create(
      <MemoryRouter initialEntries={[ '/' ]}>
        <App />
      </MemoryRouter>
    );

    let tree = target.toJSON();
    expect(tree).toMatchSnapshot();

  });

  it('renders the detail route', () => {
    const middlewares = [thunk]
    const mockStore = configureMockStore(middlewares)
    const store = mockStore({
      detail: { data: {}, error: '' }
    })

    const match = { params: { id: '1234' } };
    const detailTarget = renderer.create(
      <MemoryRouter initialEntries={[ '/detail/1234' ]}>
        <Provider store={store}>
          <DetailComponents match={ match }/>
        </Provider>
      </MemoryRouter>
    );

    let detailTree = detailTarget.toJSON();
    expect(detailTree).toMatchSnapshot();
  });
});
