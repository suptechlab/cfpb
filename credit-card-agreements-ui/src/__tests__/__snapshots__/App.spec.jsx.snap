// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state renders the detail route 1`] = `
<main
  role="main"
>
  <section
    className="card-container learning"
  >
    <section
      className="light-box"
    >
      <div
        className="loading-box"
      >
        <span
          className="cf-icon cf-icon-update cf-icon__before cf-icon__spin"
        />
        <span>
          This page is loading
        </span>
      </div>
    </section>
  </section>
</main>
`;

exports[`initial state renders without crashing 1`] = `
<main
  className="content content__1-3"
  role="main"
>
  <div
    className="wrapper content_wrapper"
  >
    <div
      className="content_main"
    >
      <header
        className="content_hero"
      >
        <h1
          className="content-header"
        >
          Credit Card Agreement Database
        </h1>
        <p>
          If you are an issuer, email 
          <a
            href="mailto:CardAgreements@consumerfinance.gov"
          >
            CardAgreements@consumerfinance.gov
          </a>
           for agreement submission instructions. Sapien justo in libero. Vestibulum mollis mauris enim. Morbi euismod magna ac lorem rutrum elementum. Donec viverra auctor lobortis. Pellentesque eu est a nulla placerat dignissim. Morbi a enim in magna semper bibendum. Etiam scelerisque, nunc ac egestas consequat, odio nibh euismod nulla, eget auctor orci nibh vel nisi.
        </p>
      </header>
    </div>
    <aside
      className="content_sidebar o-sidebar-content"
    >
      <div
        className="block"
      >
        <div
          className="m-related-links"
        >
          <header
            className="m-slug-header"
          >
            <h2
              className="a-heading"
            >
              More on the credit card agreement database
            </h2>
          </header>
          <p>
            View an archive of agreements submitted to the CFPB prior to March 2018.
          </p>
          <p>
            <a
              href="https://www.consumerfinance.gov/credit-cards/agreements/#dwnld"
            >
              View the archives
            </a>
          </p>
          <p>
            Credit card agreement database disclaimer
          </p>
          <p>
            <a
              href="#"
            >
              Read credit card database disclaimer
            </a>
          </p>
        </div>
      </div>
    </aside>
  </div>
  <div
    className="content_wrapper"
  >
    <aside
      className="content_sidebar"
    >
      <section
        className="filter-panel"
      >
        <h3>
          Filter results by...
        </h3>
        <section>
          <div
            className="o-expandable_header o-expandable_target"
          >
            <h4
              className="o-expandable_header-left o-expandable_label"
            >
              Credit card name
            </h4>
          </div>
          
          <section
            className="typeahead "
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              className="m-btn-inside-input input-contains-label"
            >
              <label
                className="input-contains-label_before input-contains-label_before__search"
              />
              <input
                autoComplete="off"
                className="a-text-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Enter card name"
                type="text"
                value=""
              />
            </div>
          </section>
          <ul />
        </section>
        <hr />
        <section>
          <div
            className="o-expandable_header o-expandable_target"
          >
            <h4
              className="o-expandable_header-left o-expandable_label"
            >
              Issuer name
            </h4>
          </div>
          
          <section
            className="typeahead "
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <div
              className="m-btn-inside-input input-contains-label"
            >
              <label
                className="input-contains-label_before input-contains-label_before__search"
              />
              <input
                autoComplete="off"
                className="a-text-input"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Enter card issuer name"
                type="text"
                value=""
              />
            </div>
          </section>
          <ul />
        </section>
      </section>
    </aside>
    <section
      className="results-panel content_main"
    >
      <summary
        className="action-bar"
      >
        <div>
          <h4>
            Showing 
            <span>
              0
            </span>
             matched results
          </h4>
        </div>
      </summary>
      <div />
      <h2>
        No results were found for your search
      </h2>
      <section
        className="light-box"
      >
        <div
          className="loading-box"
        >
          <span
            className="cf-icon cf-icon-update cf-icon__before cf-icon__spin"
          />
          <span>
            This page is loading
          </span>
        </div>
      </section>
    </section>
  </div>
</main>
`;
